src/types/mod.rs

```diff
@@ -424,6 +424,62 @@ impl ValueInternal for JsNumber {
     }
 }
 
+/// A JavaScript Date object.
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct JsDate(raw::Local);
+
+impl Value for JsDate { }
+
+impl Managed for JsDate {
+    fn to_raw(self) -> raw::Local { self.0 }
+
+    fn from_raw(_: Env, h: raw::Local) -> Self { JsDate(h) }
+}
+
+impl JsDate {
+    pub const MIN_VALID_VALUE: f64 = -8640000000000000.0;
+    pub const MAX_VALID_VALUE: f64 = 8640000000000000.0;
+
+    #[cfg(feature = "napi-runtime")]
```

Comment:
This feature flag should be around the top level declarations instead of the individual methods. IOW, the `JsDate` type shouldn't even exist unless the `"napi-runtime"` feature is enabled.